contractsV5/KyberNetwork.sol

```diff
@@ -422,62 +423,61 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
+        if (reserveArr.length == 0) return (reserves[bestReserve], 0, false);
 
-        uint[] memory destAmounts = new uint[](reserveArr.length);
+        uint[] memory rates = new uint[](reserveArr.length);
         uint[] memory reserveCandidates = new uint[](reserveArr.length);
-        uint i;
-        uint rate;
-        uint srcAmountWithFee;
-
-        for (i = 0; i < reserveArr.length; i++) {
-            //list all reserves that have this token.
-
-            srcAmountWithFee = (src == ETH_TOKEN_ADDRESS && isFeeLessReserve[reserveArr[i]])? srcAmount : 
-                srcAmount * (BPS - takerFeeBps) / BPS;
-            rate = (IKyberReserve(reserveArr[i])).getConversionRate(
-                src, 
-                dest, 
-                srcAmountWithFee, 
+        uint destAmountForComparison;
+        uint fullDestAmount;
+
+        for (uint i = 0; i < reserveArr.length; i++) {
+            //list all reserves that support this token.
+            isPayingFees = isFeePayingReserve[address(reserveArr[i])]; //not feeless
+            rates[i] = reserveArr[i].getConversionRate(
+                src,
+                dest,
+                (src == ETH_TOKEN_ADDRESS && isPayingFees) ? srcAmount * (BPS - takerFeeBps) / BPS : srcAmount, //srcAmountWithFee
                 block.number);
 
-            destAmounts[i] = srcAmountWithFee * rate / PRECISION;
-            destAmounts[i] = (dest == ETH_TOKEN_ADDRESS && isFeeLessReserve[reserveArr[i]])? destAmounts[i] : 
-                destAmounts[i] * (BPS - takerFeeBps) / BPS;
-            
-            if (destAmounts[i] > bestDestAmount) {
+            fullDestAmount = srcAmount * rates[i] / PRECISION;
+             //if token -> ETH swap & reserve is not feeless, deduct fee from destAmount
+            destAmountForComparison = (dest == ETH_TOKEN_ADDRESS && isPayingFees) ? fullDestAmount * (BPS - takerFeeBps) / BPS : fullDestAmount;
+
+            if (destAmountForComparison > bestDestAmount) {
                 //best rate is highest rate
-                bestDestAmount = destAmounts[i];
+                bestDestAmount = destAmountForComparison;
                 bestReserve = i;
             }
         }
 
-        if(bestDestAmount == 0) return (address(reserves[bestReserve]), 0, false);
+        if(bestDestAmount == 0) return (reserves[bestReserve], 0, false);
         
         reserveCandidates[0] = bestReserve;
         
         // if this reserve pays fee its acutal rate is less. so smallestRelevantRate is smaller.
         uint smallestRelevantDestAmount = bestDestAmount * BPS / (10000 + negligibleRateDiff);
 
-        for (i = 0; i < reserveArr.length; i++) {
-            
+        for (uint i = 0; i < reserveArr.length; i++) {
+
             if (i == bestReserve) continue;
-            
-            if (destAmounts[i] > smallestRelevantDestAmount) 
-            {
+
+            fullDestAmount = srcAmount * rates[i] / PRECISION;
```

Comment:
must use calcDestQty
again my fault its not here.

Reply:
actually maybe its valid not to use it if we are consistent?
but if it is should be clarified some how.

Reply:
My opinion is that it's valid not to, since we are only using the dest amount for comparison. Running issues with no. of local variables also....